src/endpoint/s3/s3_utils.js

```diff
@@ -668,6 +671,109 @@ function response_field_encoder_url(value) {
     return new URLSearchParams({ 'a': value }).toString().slice(2); // slice the leading 'a='
 }
 
+class S3SelectStream extends Transform {
+
+    static prelude_length_bytes = 12;
+    static crc_length_bytes = 4;
+
+    //end message is always the same, so we allocate and init it once.
+    static end_message = Buffer.concat([
+    Buffer.from([0, 0, 0, 56]), //PRELUDE begins: total byte length
+    Buffer.from([0, 0, 0, 40]), //headers byte length
+    Buffer.from([0xc1, 0xc6, 0x84, 0xd4]), //crc of above (end of PRELUDE)
+    Buffer.from([11]), //DATA begins, HEADERS begin: byte length of below header name
+    Buffer.from(':event-type'),
+    Buffer.from([7, 0, 3]), //type (7, string) of below header value
+    Buffer.from('End'),
+    Buffer.from([13]), //again for second header
+    Buffer.from(':message-type'),
+    Buffer.from([7, 0, 5]),
+    Buffer.from('event'), //end of HEADERS
+    //no PAYLOAD (DATA ends)
+    Buffer.from([0xfe, 0x2c, 0xee, 0x99]) //MESSAGE CRC (crc of above)
+    ]);
+
+//same for headers of records message
```

Comment:
I think an indentation is missing here (as I saw in your code style in the lines before).